src/wallet/signer.rs

```diff
@@ -229,19 +299,25 @@ impl Signer for DescriptorXKey<ExtendedPrivKey> {
             return Ok(());
         }
 
+        let tap_key_origins = psbt.inputs[input_index]
+            .tap_key_origins
+            .iter()
+            .map(|(pk, (_, keysource))| (SinglePubKey::XOnly(*pk), keysource));
         let (public_key, full_path) = match psbt.inputs[input_index]
             .bip32_derivation
             .iter()
-            .filter_map(|(pk, &(fingerprint, ref path))| {
-                if self.matches(&(fingerprint, path.clone()), secp).is_some() {
-                    Some((pk, path))
+            .map(|(pk, keysource)| (SinglePubKey::FullKey(PublicKey::new(*pk)), keysource))
+            .chain(tap_key_origins)
+            .filter_map(|(pk, keysource)| {
+                if self.matches(keysource, secp).is_some() {
+                    Some((pk, keysource.1.clone()))
                 } else {
                     None
                 }
             })
             .next()
         {
-            Some((pk, full_path)) => (pk, full_path.clone()),
+            Some((pk, full_path)) => (pk, full_path),
             None => return Ok(()),
         };
```

Comment:
Collecting allocates a new vec but we really only need the first item, so that would be a bit more wasteful.

I'll see if I can write this down more clearly.

Reply:
Ok I switched to `find_map()` which directly returns the first non-`None` item, instead of mapping the iterator and taking the first item.

Reply:
Yes.. Thats even more better..